libs/server/Resp/RespCommand.cs

```diff
@@ -120,602 +146,1200 @@ enum RespCommandOption : byte
-                        return (RespCommand.BITFIELD, 0);
-
-                    // -----8-----  ---8---  --2--
-                    // [$11\r\nBIT | FIELD_RO | \r\n$]
-                    if (*(long*)ptr == 6073458183424258340L && *(long*)(ptr + 8) == 5715735624028604742L && *(short*)(ptr + 8 + 8) == 2573)
-                        return (RespCommand.BITFIELD_RO, 0);
-
-                    // $5\r\nBITO | P\r\n$ | [2|3]\r\n[AND|OR|XOR|NOT|]\r\n | $4\r\ndest $4\r\nsrc1\r\n $4\r\nsrc2\r\n
-                    if (*(long*)ptr == 5716274375025308964L)
+                    if (*(recvBufferPtr + readHead) != '*')
                     {
-                        // ----8----    -----8-----   ---4---
-                        // $5\r\nBITO | P\r\n$3\r\nA | ND\r\n
-                        // $5\r\nBITO | P\r\n$3\r\na | nd\r\n
-                        if ((*(long*)(ptr + 8) == 4686572875531554128L && *(int*)(ptr + 16) == 168641614) ||
-                            (*(long*)(ptr + 8) == 6992415884745248080L && *(int*)(ptr + 16) == 168649838))
-                            return (RespCommand.BITOP, (byte)BitmapOperation.AND);
-
-                        // ----8----    -----8-----   ---4---
-                        // $5\r\nBITO | P\r\n$2\r\nO | R\r\n$
-                        // $5\r\nBITO | P\r\n$2\r\no | r\r\n$
-                        if ((*(long*)(ptr + 8) == 5695379187767577936L && *(int*)(ptr + 16) == 604638546) ||
-                            (*(long*)(ptr + 8) == 8001222196981271888L && *(int*)(ptr + 16) == 604638578))
-                            return (RespCommand.BITOP, (byte)BitmapOperation.OR);
-
-                        // ----8----    -----8-----   ---4---
-                        // $5\r\nBITO | P\r\n$3\r\nX | OR\r\n
-                        // $5\r\nBITO | P\r\n$3\r\nx | or\r\n
-                        if ((*(long*)(ptr + 8) == 6343897538403896656L && *(int*)(ptr + 16) == 168645199) ||
-                            (*(long*)(ptr + 8) == 8649740547617590608L && *(int*)(ptr + 16) == 168653423))
-                            return (RespCommand.BITOP, (byte)BitmapOperation.XOR);
-
-                        // ----8----    -----8-----   ---4---
-                        // $5\r\nBITO | P\r\n$3\r\nN | OT\r\n
-                        if ((*(long*)(ptr + 8) == 5623321598024617296L && *(int*)(ptr + 16) == 168645711) ||
-                            (*(long*)(ptr + 8) == 7929164607238311248L && *(int*)(ptr + 16) == 168653935))
-                            return (RespCommand.BITOP, (byte)BitmapOperation.NOT);
+                        // TODO: Error Handling should be done outside of parsing function.
+                        if (bytesRead - readHead < 2)
+                        {
+                            success = false;
+                            return (RespCommand.INVALID, subCmd);
+                        }
+
+                        if (*(ushort*)(recvBufferPtr + bytesRead - 2) != 2573)
```

Comment:
nit: Might want to add a comment explaining the error conditions we're checking here

Reply:
This has been addressed by simplifying this function.